sympy/solvers/solveset.py

```diff
@@ -1088,23 +1126,19 @@ def _is_two_term_exponential(f, symbol):
 
     * Philosophy behind the helper
 
-    The function checks for all two terms being in power form
-    with `symbol` in their exponents but not in their base.
+    The function extracts each term of the equation and checks if it is
+    of exponential form w.r.t `symbol`.
     """
-
-    if not (isinstance(f, Add) and len(f.args) == 2):
-        # solving for the sum of more than two powers is possible
-        # but not yet implemented
-        return False
-
-    a, b = f.args
-    a_coeff, a_exp_term = a.as_independent(symbol)
-    b_coeff, b_exp_term = b.as_independent(symbol)
-
-    return all(isinstance(expr_arg, (Pow, exp)) and (
-        symbol in expr_arg.exp.free_symbols and
-        symbol not in expr_arg.base.free_symbols)
-        for expr_arg in (a_exp_term, b_exp_term))
+    expr_args = _term_factors(f)
+    for expr_arg in expr_args:
+        if (isinstance(expr_arg.evalf(), Number) or
```

Comment:
the constant terms or variables (not `symbol`) will be either the coefficients or a constant which will not affect the definition of what exponential equations are.

```python

>>> _is_exponential(2**x + 3**x - 5, x)
True
>>>  _is_exponential(2**x + 3**x - y, x)
True
>>>  _is_exponential(2**x + 3**x - x, x)    ## a lambert type
False
```
@smichr Is the approach of the helper okay as required?

Reply:
I think the logic here should be:

```python
rv = False
for a in _term_factors(expr):
    if symbol not in a.free_symbols:
        continue
    if isinstance(a, (Pow, exp)
            ) and symbol not in a.base.free_symbols and symbol in a.exp.free_symbols:
        rv = True
    else:
        return False  # depended on symbol in non-exponential way
return rv
```

I advise against using evalf to see if something is a Number for two reasons: it might evaluate to `1 + 2*I` which is not a number and it is more expensive to evaluate (in general) than to check free symbols.

Reply:
The mods you made still allow `_is_exponential(42)` to return True. I know you don't plan on sending something without x in it to this routine, but it's easy to make it fool proof so it can be used regardless of what simplification may have reduced the expression to a constant when this function is eventually used.

Reply:
Done :+1:

Reply:
Did you consider my last comment made jsut a few minutes before yours? No mods of that sort have been made.

Reply:
Oh sorry I didn't realize that your comment was made.

>  but it's easy to make it fool proof so it can be used regardless of what simplification may have reduced the expression to a constant when this function is eventually used.

Okay I will modify as you suggested

Reply:
Or should I just add a condition of `f` having  the `symbol`

```python

def _is_exponential(f, symbol):
    if symbol not in f.free_symbol:
        return False
   # rest as it is in the recent commit
```

Reply:
At worst, this will require a double walk of the expression looking for free symbols. I think it's better to do the scans a piece at a time.

Reply:
Okay cool. I will modify the changes to the one you suggested.

Apart from this what else needs to be looked into? or is this PR in a good shape?